mmdet3d/models/dense_heads/mlp_bbox_head.py

```diff
@@ -0,0 +1,123 @@
+from mmcv.cnn import ConvModule
+from torch import nn as nn
+
+from mmdet.models.builder import HEADS
+
+
+@HEADS.register_module()
+class MLPBBoxHead(nn.Module):
+    r"""More general bbox head, with shared mlp layers and two optional
+    separated branches.
+
+    .. code-block:: none
+
+                   /-> cls mlps -> cls_score
+        shared mlps
+                   \-> reg mlps -> bbox_pred
+    """
+
+    def __init__(self,
+                 in_channels=0,
+                 shared_mlp_channels=(),
+                 cls_mlp_channels=(),
+                 num_cls_out_channels=0,
+                 reg_mlp_channels=(),
+                 num_reg_out_channels=0,
+                 conv_cfg=dict(type='Conv1d'),
+                 norm_cfg=dict(type='BN1d'),
+                 act_cfg=dict(type='ReLU'),
+                 bias='auto',
+                 *args,
+                 **kwargs):
+        super(MLPBBoxHead, self).__init__(*args, **kwargs)
+        assert in_channels > 0
+        assert num_cls_out_channels > 0
+        assert num_reg_out_channels > 0
+        self.in_channels = in_channels
+        self.shared_mlp_channels = shared_mlp_channels
+        self.cls_mlp_channels = cls_mlp_channels
+        self.num_cls_out_channels = num_cls_out_channels
+        self.reg_mlp_channels = reg_mlp_channels
+        self.num_reg_out_channels = num_reg_out_channels
+        self.conv_cfg = conv_cfg
+        self.norm_cfg = norm_cfg
+        self.act_cfg = act_cfg
+        self.bias = bias
+
+        # add shared convs and fcs
+        if len(self.shared_mlp_channels) > 0:
+            self.shared_mlps = self._add_mlp_branch(self.in_channels,
+                                                    self.shared_mlp_channels)
+            out_channels = self.shared_mlp_channels[-1]
+        else:
+            out_channels = self.in_channels
+
+        # add cls specific branch
+        prev_channel = out_channels
+        if len(self.cls_mlp_channels) > 0:
+            self.cls_mlps = self._add_mlp_branch(prev_channel,
+                                                 self.cls_mlp_channels)
+            prev_channel = self.cls_mlp_channels[-1]
+
+        self.cls_pred = nn.Conv1d(prev_channel, num_cls_out_channels, 1)
```

Comment:
Use `build_conv_layer` API with conv_cfg here thus we could also use `Conv2d` in this head.